pkg/option/config.go

```diff
@@ -2888,7 +2882,7 @@ func (c *DaemonConfig) Validate(vp *viper.Viper) error {
 	if c.ClusterID != 0 {
```

Comment:
Since we have a new type here we could perhaps encapsulate validation to ensure the values are legit vs externalizing it?

Reply:
Makes sense, I've moved the validation logic as methods of the `ClusterInfo` struct.